File: Modules/SMES/Esop/SIE.Web.ESop/Displays/Commands/InitESopCommand.cs
using SIE.ESop.Displays;
using SIE.Web.Command;

namespace SIE.Web.ESop.Displays.Commands
{
    /// <summary>
    /// 初始化ESOP
    /// </summary>
    public class InitESopCommand : ViewCommand
    {
        /// <summary>
        /// 执行命令
        /// </summary>
        /// <param name="args">参数</param>
        /// <param name="scope">作用域</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            RT.Service.Resolve<DisplayPointController>().InitEsop();
            return true;
        }
    }
}

File: Modules/SMES/Esop/SIE.Web.ESop/Displays/Commands/InitESopCommand.js
SIE.defineCommand('SIE.Web.ESop.Displays.Commands.InitESopCommand', {
    meta: { text: "初始化ESOP", group: "edit", iconCls: "icon-Initialization icon-blue" },
    canExecute: function (view) {
        return true;
    },
    execute: function (view, source) {
        var me = this;
        me.view.execute({
            data: {},
            success: function (res) {
                if (res.Result)
                    SIE.Msg.showMessage(res.Result);
                else
                    SIE.Msg.showInstantMessage('初始化成功'.t());
                me.view.loadData();
            },
        }, me.view);
    },
});